# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/c3c/no_macho.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

index 8958c74..109d66d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -218,7 +218,6 @@ if(C3_WITH_LLVM)
         find_library(LLD_COFF NAMES liblldCOFF.dylib lldCOFF.lib lldCOFF.a liblldCOFF.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_COMMON NAMES liblldCommon.dylib lldCommon.lib lldCommon.a liblldCommon.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_ELF NAMES liblldELF.dylib lldELF.lib lldELF.a liblldELF.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
-        find_library(LLD_MACHO NAMES liblldMachO.dylib lldMachO.lib lldMachO.a liblldMachO.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_MINGW NAMES liblldMinGW.dylib lldMinGW.lib lldMinGW.a liblldMinGW.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_WASM NAMES liblldWasm.dylib lldWasm.lib lldWasm.a liblldWasm.a PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
     else()
@@ -230,7 +229,6 @@ if(C3_WITH_LLVM)
         find_library(LLD_COFF NAMES liblldCOFF.so PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_COMMON NAMES liblldCommon.so PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_ELF NAMES liblldELF.so PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
-        find_library(LLD_MACHO NAMES liblldMachO.so PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_MINGW NAMES liblldMinGW.so PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
         find_library(LLD_WASM NAMES liblldWasm.so PATHS ${LLVM_LIBRARY_DIRS} NO_DEFAULT_PATH)
     endif()
@@ -249,7 +247,6 @@ if(C3_WITH_LLVM)
             ${LLD_WASM}
             ${LLD_MINGW}
             ${LLD_ELF}
-            ${LLD_MACHO}
     )
 
     if (APPLE)
index 81ee9a0..faf9a98 100644
--- a/wrapper/src/wrapper.cpp
+++ b/wrapper/src/wrapper.cpp
@@ -49,14 +49,6 @@ namespace lld {
 		LINK_SIG
 	}
 
-	namespace mach_o {
-		LINK_SIG
-	}
-
-	namespace macho {
-		LINK_SIG
-	}
-
 	namespace wasm {
 		LINK_SIG
 	}
@@ -107,8 +99,8 @@ static bool llvm_link(ObjFormat format, const char **args, int arg_count, const
 			success = lld::elf::link(CALL_ARGS);
 			break;
 		case MACHO:
-			success = lld::macho::link(CALL_ARGS);
-			break;
+			printf("MachO not enabled");
+			exit(1);
 		case WASM:
 			success = lld::wasm::link(CALL_ARGS);
 			break;
